### PR TITLE
Add description about check constraint

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/14227
- Other reference link(s):
 https://github.com/pingcap/tidb/issues/41711
### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: constraints.md [hunk 1]
```diff
@@ -52,20 +52,81 @@ Query OK, 1 row affected (0.03 sec)
...
+    CREATE TABLE t(a INT CHECK(a > 10) NOT ENFORCED, b INT, c INT, CONSTRAINT c1 CHECK (b > c));
+    ```
+
+- Example of adding a `CHECK` constraint using the `ALTER TABLE` statement:
+
+    ```sql
+    ALTER TABLE t ADD CONSTRAINT CHECK (1 < c);
+    ```
+
+When adding or enabling a `CHECK` constraint, TiDB validates the existing data in the table. If any data violates the constraint, the operation of adding `CHECK` constraint will fail and return an error.
```
Comment (qiancai) [inline constraints.md:86]:
```suggestion
When adding or enabling a `CHECK` constraint, TiDB validates the existing data in the table. If any data violates the constraint, the operation of adding the `CHECK` constraint will fail and return an error.
```

File: constraints.md [hunk 2]
```diff
@@ -52,20 +52,81 @@ Query OK, 1 row affected (0.03 sec)
...
+
+When adding a `CHECK` constraint, you can either specify a constraint name or leave the name unspecified. If no constraint name is specified, TiDB automatically generates a constraint name in the `<tableName>_chk_<1, 2, 3...>` format.
+
+### Viewing `CHECK` constraints
```
Comment (qiancai) [inline constraints.md:90]:
```suggestion
### View `CHECK` constraints
```

**Review by fzzf678 — APPROVED** [review]

File: constraints.md [hunk 3]
```diff
@@ -52,20 +52,81 @@ Query OK, 1 row affected (0.03 sec)
 
 ## CHECK
 
-TiDB parses but ignores `CHECK` constraints. This is MySQL 5.7 compatible behavior.
+A `CHECK` constraint is used to restrict the values of a field in a table to meet your specified conditions. When the `CHECK` constraint is added to a table, TiDB checks whether the constraint is satisfied during the insertion or update of data into the table. If the constraint is not met, an error is returned.
```
Comment (qiancai) [inline constraints.md:55]:
```suggestion
A `CHECK` constraint is used to restrict the values of a field in a table to meet your specified conditions. When the `CHECK` constraint is added to a table, TiDB checks whether the constraint is satisfied during the insertion or updates of data into the table. If the constraint is not met, an error is returned.
```

File: constraints.md [hunk 4]
```diff
@@ -52,20 +52,81 @@ Query OK, 1 row affected (0.03 sec)
...
+```sql
+ALTER TABLE t DROP CONSTRAINT t_chk_1;
+```
+
+### Enable or disable `CHECK` constraints
+
+When [adding a `CHECK` constraint](#add-check-constraints) to a table, you can specify whether TiDB needs to implement the constraint check during data insertion or updates.
+
+- If `NOT ENFORCED` is specified, TiDB does not validate the constraint conditions during data insertion or updates.
+- If `NOT ENFORCED` is not specified or `ENFORCED` is specified, TiDB validates the constraint conditions during data insertion or updates.
```
Comment (Oreoxmt) [inline constraints.md:124]:
```suggestion
- If `NOT ENFORCED` is specified, TiDB does not check the constraint conditions during data insertion or updates.
- If `NOT ENFORCED` is not specified or `ENFORCED` is specified, TiDB checks the constraint conditions during data insertion or updates.
```

File: constraints.md [hunk 5]
```diff
@@ -52,20 +52,81 @@ Query OK, 1 row affected (0.03 sec)
...
+    CREATE TABLE t(a INT CHECK(a > 10) NOT ENFORCED, b INT, c INT, CONSTRAINT c1 CHECK (b > c));
+    ```
+
+- Example of adding a `CHECK` constraint using the `ALTER TABLE` statement:
+
+    ```sql
+    ALTER TABLE t ADD CONSTRAINT CHECK (1 < c);
+    ```
+
+When adding or enabling a `CHECK` constraint, TiDB validates the existing data in the table. If any data violates the constraint, the operation of adding the `CHECK` constraint will fail and return an error.
```
Comment (Oreoxmt) [inline constraints.md:86]:
```suggestion
When adding or enabling a `CHECK` constraint, TiDB checks the existing data in the table. If any data violates the constraint, the operation of adding the `CHECK` constraint will fail and return an error.
```

Comment (Oreoxmt) [reply constraints.md:86]: Keep it consistent with "When the `CHECK` constraint is added to a table, TiDB **checks** whether the constraint is satisfied during the insertion or updates of data into the table. "

File: constraints.md [hunk 6]
```diff
@@ -52,20 +52,81 @@ Query OK, 1 row affected (0.03 sec)
 
 ## CHECK
 
-TiDB parses but ignores `CHECK` constraints. This is MySQL 5.7 compatible behavior.
+A `CHECK` constraint is used to restrict the values of a field in a table to meet your specified conditions. When the `CHECK` constraint is added to a table, TiDB checks whether the constraint is satisfied during the insertion or updates of data into the table. If the constraint is not met, an error is returned.
```
Comment (Oreoxmt) [inline constraints.md:55]:
```suggestion
A `CHECK` constraint restricts the values of a column in a table to meet your specified conditions. When the `CHECK` constraint is added to a table, TiDB checks whether the constraint is satisfied during the insertion or updates of data into the table. If the constraint is not met, an error is returned.
```

**Review by Oreoxmt — APPROVED** [review]
/lgtm

**Review by hfxsd — APPROVED** [review]

**Pull request merged**